src/server/svr_jobfunc.c

```diff
@@ -554,7 +554,8 @@ svr_dequejob(job *pjob)
 
 		/* update any entity count and entity resources usage at que */
 
-		account_entity_limit_usages(pjob, pque, NULL, DECR);
+		account_entity_limit_usages(pjob, pque, NULL, DECR,
+				pjob->ji_etlimit_decr_queued ? ETLIM_ACC_ALL_MAX : ETLIM_ACC_ALL);
```

Comment:
if `pjob->ji_etlimit_decr_queued` is set, then we now have to `DECR` only `set_entity_*_sum_max` as `set_entity_*_sum_queued` was already decremented when the job entered run. Hence passing `ETLIM_ACC_ALL_MAX`

if `pjob->ji_etlimit_decr_queued` is unset, then we have to decrement all the 4 limit attributes. Hence passing `ETLIM_ACC_ALL`

Reply:
Is this called for only subjobs or all normal jobs? I thought job_purge will DECR the limit for all jobs anyways. Or maybe you need this only when job is moved to another queue not when job is being purged?

Reply:
This is for both subjobs and normal jobs,
`job_purge()` will decrement the server level usage,
Here we will decrement the queue level usage.